pkg/syncer/cnsoperator/controller/cnsfileaccessconfig/cnsfileaccessconfig_controller.go

```diff
@@ -321,6 +349,90 @@ func (r *ReconcileCnsFileAccessConfig) Reconcile(request reconcile.Request) (rec
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileCnsFileAccessConfig) configureFileVolumeACL(ctx context.Context, volumeID string, vm *vmoperatortypes.VirtualMachine, instance *cnsfileaccessconfigv1alpha1.CnsFileAccessConfig) error {
+	log := logger.GetLogger(ctx)
+	networkProvider, err := util.GetNetworkProvider(ctx)
+	if err != nil {
+		msg := fmt.Sprintf("Failed to identify the network provider. Error: %+v", err)
+		log.Error(msg)
+		return errors.New(msg)
+	}
+	var nsxConfiguration bool
+	if networkProvider != "" {
+		if networkProvider == util.NSXTNetworkProvider {
+			nsxConfiguration = true
+		} else if networkProvider == util.VDSNetworkProvider {
+			nsxConfiguration = false
+		} else {
+			msg := fmt.Sprintf("Unknown network provider. Error: %+v", err)
+			log.Error(msg)
+			return errors.New(msg)
+		}
+	}
+	tkgVMIP, ok := externalIPtoClientVM[vm.Name]
```

Comment:
Lets just call `GetTKGVMIP()` everytime to get the  IP address and make sure the code is functionally correct. We can add a cache as a future enhancement. What do you think?

Reply:
I agree with this. I have added a TODO for now